client/client_test.go

```diff
@@ -168,15 +159,16 @@ func (s *TestSuite) TestChannels(t *C) {
 	// Start send/recv chans, but idle until successful Connect.
 	ws.Start()
 	defer ws.Stop()
+	defer ws.Disconnect()
```

Comment:
Turns out there is some race condition issue that makes this test fail in go 1.8 - it hangs forever. Ensuring that `ws.Disconnect` is run before `ws.Stop` helps. Let's hope this is only in tests, not in real life code.